[PLUGIN-1672] Fixed unit tests failures

Aryan-Verma opened the pull request:
BUT tests in develop were failing after the recent change [PR](https://github.com/data-integrations/google-cloud/pull/1281). Fixed DataplexBatchSourceConfigTest unit test. Also fixed error message in e2e tests for incorrect table name scenario.

Related Jira ticket: [PLUGIN-1672](https://cdap.atlassian.net/browse/PLUGIN-1672)

[PLUGIN-1672]: https://cdap.atlassian.net/browse/PLUGIN-1672?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

itsankit-google commented on src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java:
```diff
@@ -676,7 +676,7 @@ public static void validateDataset(String dataset, String datasetPropertyName, F
   public static void validateTable(String table, String tablePropertyName, FailureCollector collector) {
     // Allowed character validation for table name as per https://cloud.google.com/bigquery/docs/tables
-    String errorMessage = "Table name can only contain letters (lower or uppercase), numbers, '_' and '-'.";
+    String errorMessage = "Table name can only contain letters (lower or uppercase), numbers and '_'.";
```
Please revert this change, according to bigquery public documentation: https://cloud.google.com/bigquery/docs/tables#table_naming

BQ table names can have hyphens (`-`) why are we making this change, was it done unknowingly?

Aryan-Verma replied:
Reverted. It was done by mistake

itsankit-google reviewed (changes requested):
Please add a JIRA to the PR description since it was missed in the original PR.

Aryan-Verma commented:
> Please add a JIRA to the PR description since it was missed in the original PR.

Added the Jira ticket link. I have also added the link in the description of the original PR
